### PR TITLE
Validate URL scheme for Auth

### PR DESCRIPTION
**Pull request by jaelim-ms**
* [X] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you add unit tests?
* [X] Did you check UI tests on the sample app? They are not executed on CI.
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add custom URL scheme validation on Auth start.

## Related PRs or issues

[AB#62677](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/62677)

### CODE REVIEW

File: AppCenterAuth/AppCenterAuth/MSAuth.m [hunk 1]
```diff
@@ -258,6 +262,22 @@ - (void)signOut {
 
 #pragma mark - Private methods
 
+- (BOOL)checkURLSchemeRegistered:(NSString *)urlScheme {
+  NSArray *schemes;
+  NSString *typeRole;
+  NSArray *types = [MS_APP_MAIN_BUNDLE objectForInfoDictionaryKey:kMSCFBundleURLTypes];
```
Comment (achocron) [inline AppCenterAuth/AppCenterAuth/MSAuth.m]: I guess this will crash if it is not an `NSArray`

File: AppCenterAuth/AppCenterAuth/MSAuth.m [hunk 2]
```diff
@@ -258,6 +262,22 @@ - (void)signOut {
...
+  for (NSDictionary *urlType in types) {
+    schemes = urlType[kMSCFBundleURLSchemes];
```
Comment (achocron) [inline AppCenterAuth/AppCenterAuth/MSAuth.m]: I think this will crash if `urlType` is not an `NSDictionary`, and same if `scheme` and `typeRole` are not `NSString`

File: AppCenterAuth/AppCenterAuth/MSAuth.m (hunk 1 above)
Comment (achocron) [reply AppCenterAuth/AppCenterAuth/MSAuth.m]: Also can it be `nil`?

File: AppCenterAuth/AppCenterAuth/MSAuth.m (hunk 2 above)
Comment (achocron) [reply AppCenterAuth/AppCenterAuth/MSAuth.m]: Or if `scheme` is not an `NSArray`. Basically, should we type check these?

Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/MSAuth.m]: `CFBundleURLTypes` is an array of dictionaries. [Apple Doc](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleurltypes)
`CFBundleURLSchemes` is an array of strings. [Apple Doc](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleurltypes/cfbundleurlschemes)

File: AppCenterAuth/AppCenterAuth/MSAuth.m (hunk 1 above)
Comment (jaelim-ms) [reply AppCenterAuth/AppCenterAuth/MSAuth.m]: It is an `NSArray`. Look at below answer.
I also added a unit test for the case if it is `nil`.

**Comment by jaelim-ms** [comment]
@achocron I don't think we need to consider that invalid case in the code.
